### DISCUSSION TITLE
v3.0.0

### DISCUSSION THREAD

**Discussion by JLyne** [JLyne/LiveAtlas#396] [General]
Today's final 3.0.0 release brings [Overviewer ](https://github.com/overviewer/Minecraft-Overviewer) map support, a searchable marker list in the sidebar and various UI improvments and bug fixes.

# Changes since 3.0.0-beta.4
## Added
 - Added `disableMarkerUI` index.html option for disabling the new marker list

## Misc.
 - Updated to latest leaflet

# Changes since 2.0.1
## Added
 - **Marker list**
   - New sidebar section between maps and players
   - Can be toggled with `CTRL + SHIFT + I` (`m` was taken by maps)
   - Initially shows marker sets with marker counts underneath. Clicking a set will list its contents.
   - Dynmap markers will show their `label` in the list; squaremap/pl3xmap markers will show their `tooltip`. HTML is stripped.
   - Markers will show their location and icon in the list
     - Point markers will show their exact location, other marker types will show the center of all their points
     - Markers without icons will use a generic icon of their type
   - Markers without a name will show a fallback message defined in index.html, but will still be displayed and can be panned to.
   - The marker list for each marker set has a search bar which will filter the list based on name
   - Clicking a marker in the list will pan to it
     - Point markers will simply pan, but other marker types will also zoom to fit their shape as much as possible 
   - Keyboard navigation is fully supported.
     - Arrow keys, home/end and space/enter work as with other lists
     - Backspace whilst inside a marker list will take you back to the marker set list
     - CTRL + F in a marker list will focus the search
   - To avoid performance issues with large sets, only 50 markers will be listed at first
     - A "Show more" button will appear at the bottom to allow displaying further markers
   - Can be disabled with the `disableMarkerUI` option in index.html
 - **Overviewer support**
   - Most standard features should be fully supported, including: Overlay tilesets, markers, spawn markers, etc
   - Overviewer format URLs are partially supported. Everything except the zoom level will be handled correctly.
   - Due to overviewer config files being JS files, if you have a Content Security Policy you will need to allow `unsafe-eval` and `unsafe-inline` for scripts if you have any configured overviewer maps.
   - Modifications to overviewer are unlikely to be supported if they require any frontend changes
     - Additional injected marker scripts are supported if they add markers to the global `markersDB` or `markers` objects like the default scripts do
   - The URL location marker is not currently implemented. If this is later implemented it would likely be a toggleable feature for all map types
 - **Following improvements**
   - Following UI now sticks to the bottom of the viewport like in initial LiveAtlas versions
   - Player world and location is now shown when the player is visible
   - Clicking the location will copy it to your clipboard
   - Fixed long names breaking the layout
   - Existing player follows are now cancelled when manually switching worlds
 - **Sidebar improvements** 
   - Added tweaked map icons by [@Katuend](https://twitter.com/katuend)
   - Sidebar section headers and search bars now stick to the top of the viewport when scrolling the sidebar
   - Searching players now scrolls the list back to the top
   - Pressing down arrow on buttons for already expanded sidebar sections will now focus the section
   - Fixed long headings breaking the layout
   - Improved scrolling performance a bit
   - Increased sidebar width slightly to more comfortably fit marker names
 - Added `disableContextMenu` index.html option, to allow disabling the custom map context menu
 - Missing index.html messages will now show a fallback message instead of nothing
 - Squaremap maps can now be configured with `squaremap` as well as `pl3xmap` in index.html
 - Added support for Dynmap's new `tilescale` setting
 - Dynmap shape markers with a label will now show a tooltip with the label on hover, matching Pl3xmap behaviour.
 - Dynmap point markers will now be centered in the block they are located in, rather than displaying in the top left corner

## Fixed
 - Fixed sidebar sections attempting to focus themselves repeatedly on initial load if the UI is visible
 - Fixed player heads continuing to load for the previous server when switching servers
 - Fixed tooltips not updating if a marker updates
 - Fixed player image not updating on follow target change
 - Fixed point marker icons not updating when they are changed
 - Fixed some Squaremap/Pl3xmap markers showing their position as NaN, NaN in the marker list
 - Fixed various issues handling Dynmap's `append_to_world` setting (#345)
 - Fixes worlds appearing in the map list if all of their maps are appended to another world
 - Fixes multiple maps with the same name replacing each other when appended to the same world
 - Fixed incorrect handling of Squaremap/Pl3xmap shape markers, causing some to be incorrectly filled (#359)

## Misc.
 - Improved marker set creation performance
 - Many internal changes to marker updating and storage
 - Player markers are now only created once per player on a server.
   - This should improve world switching performance slightly at high player counts.
 - Player images are now updated if the player image URL changes (i.e switching worlds with different `heads_url`s)

<hr /><em>This discussion was created from the release <a href='https://github.com/JLyne/LiveAtlas/releases/tag/v3.0.0'>v3.0.0</a>.</em>

**Comment by smooreace** [comment]
tt seems that only chunks loaded by players appear on the map? Is it possible to force exposing a given area base on a designated center location?
